RELEASE_NOTES.md

```diff
@@ -1,3 +1,7 @@
+# Upcoming
+
+- feat: Snippet in the error messages of the language server. (https://github.com/dafny-lang/dafny/pull/2434)
```

Comment:
This description isn't entirely clear to me, and could probably do more to show how nice this is to have.

Reply:
What about the new description?